### PR TITLE
Issue #11446: Update DetailAstImplTest to use execute

### PR DESCRIPTION
**Pull request by suniti0804**
Issue #11446

### CODE REVIEW

**Review by romani — APPROVED** [review]
generated path does not need BDD style.

ok to merge if CI pass

sornar give a challange: https://sonarcloud.io/project/issues?resolved=false&severities=BLOCKER&sinceLeakPeriod=true&pullRequest=14148&id=org.checkstyle%3Acheckstyle&open=AYxtyp6aZQAQqRv7kp4y

**Comment by romani** [comment]
@suniti0804, try to update test to insert our BDD comment with config.
I do not remember how we bypassed Sonar in other cases, but for this case we can reuse BDD style easily, without harm to test.

**Comment by suniti0804** [comment]
> sornar give a challange: https://sonarcloud.io/project/issues?resolved=false&severities=BLOCKER&sinceLeakPeriod=true&pullRequest=14148&id=org.checkstyle%3Acheckstyle&open=AYxtyp6aZQAQqRv7kp4y

resolved, added an assertion in test method to assert that the number of files in the temporary directory has not increased,
meaning that no TODO comments were found in the file

File: src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java [hunk 1]
```diff
@@ -693,10 +694,15 @@ public void testManyComments() throws Exception {
...
+
         final DefaultConfiguration checkConfig = createModuleConfig(TodoCommentCheck.class);
 
-        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
-        verify(checkConfig, file.getAbsolutePath(), expected);
+        execute(checkConfig, file.getAbsolutePath());
+
+        final int finalFileCount = Objects.requireNonNull(temporaryFolder.list()).length;
+
+        Assertions.assertEquals(initialFileCount, finalFileCount);
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java:705]: No, this is too crazy, next engineer will not understand this.
Let's add header to file and use our verify with embedded config.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java (hunk 1 above)
Comment (suniti0804) [reply src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java:705]: There doesn't exist any file with the name `InputDetailASTManyComments`. 
I tried running the test with verifyWithInlineConfigParser and it mentioned the following file path - Config comment not specified properly in /var/folders/ry/ztsc86hs7wvc09v5qtvt1gtw0000gn/T/junit13776525294712844995/InputDetailASTManyComments.java

which is again nowhere to be found in the project.

Comment (nrmancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java:705]: Yes, this file is generated in this test. See the for loop above this code. We need to update this test to insert our config comment into this generated file.

Comment (suniti0804) [reply src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java]: Yes done

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**